src/core/models.py

```diff
@@ -177,12 +179,27 @@ def create(self, **kwargs):
 
 
 class AccountManager(BaseUserManager):
-    def create_user(self, email, password=None, **kwargs):
-        if not email:
-            raise ValueError('Users must have a valid email address.')
+    def create_user(self, username=None, password=None, email=None, **kwargs):
+        """ Creates a user from the given username or email
+        In Janeway, users rely on email addresses to log in. For compatibility
+        with 3rd party libraries, we allow a username argument, however only a
+        email address will be accepted as the username and email.
+        """
+        if not email and username:
+            email = username
+            if "username" in kwargs:
+                del kwargs["username"]
+        try:
+            validate_email(email)
+            self.normalize_email(email)
+        except(ValidationError, TypeError, ValueError):
+            raise ValueError(f'{email} not a valid email address.')
 
         account = self.model(
-            email=self.normalize_email(email),
+            email=email,
+            # Lowercasing the email into the username gets around duplicate
+            # accounts due to case-sensitivity. We still preserve the original
+            # casing for the email field and rely on username for CI constraint
```

Comment:
I don't entirely follow this comment, so others might have difficulty too. What does the last part mean?

Reply:
I tried to explain exisitng behaviour because I thought it wasn't clear, seems I didn't do that well. The basic idea is that we need a case-sensitive email field, but we need a case-insensitive  constraint to avoid accounts being created with variations in case e.g.: `m.SaNcHeZ@bbk.ac.uk`. The way this is achieved is by setting a unique constraint on `username` while making it case insensitive. Any ideas on how I can reword it to make it clear?

Reply:
OK, now I get it, thanks! Maybe something like this?

```
diff --git a/src/core/models.py b/src/core/models.py
index aba452e2..d480da59 100644
--- a/src/core/models.py
+++ b/src/core/models.py
@@ -196,10 +196,12 @@ class AccountManager(BaseUserManager):
             raise ValueError(f'{email} not a valid email address.')
 
         account = self.model(
+            # The original case of the email is preserved
+            # in the email field
             email=email,
-            # Lowercasing the email into the username gets around duplicate
-            # accounts due to case-sensitivity. We still preserve the original
-            # casing for the email field and rely on username for CI constraint
+            # The email is lowercased in the username field
+            # so that we can perform case-insensitive checking
+            # and avoid creating duplicate accounts
             username=email.lower(),
         )
```